index.js

```diff
@@ -344,55 +344,65 @@ PersonaClient.prototype.validateToken = function (opts, next) {
  */
-PersonaClient.prototype.validateHTTPBearerToken = function (request, response, next) {
+PersonaClient.prototype.validateHTTPBearerToken = function validateHTTPBearerToken(request, response, next) {
+    var config = {
+        token: this._getToken(request),
+        scope: request.param('scope'),
+        xRequestId: this.getXRequestId(request),
+    };
+
     if (arguments.length > 3) {
-        throw "Usage: validateHTTPBearerToken(request, response, next)";
+        throw new Error('Usage: validateHTTPBearerToken(request, response, next)');
     }
-    var token = this._getToken(request);
-    var xRequestId = this.getXRequestId(request);
 
-    this.validateToken({token: token, scope: request.param("scope"), xRequestId: xRequestId}, function (error, validationResult) {
+    function callback(error, validationResult) {
         if (!error) {
             next(null, validationResult);
             return;
         }
 
-        switch(error) {
+        switch (error) {
         case ERROR_TYPES.INVALID_TOKEN:
             response.status(401);
             response.json({
-                "error": "no_token",
-                "error_description": "No token supplied"
+                'error': 'no_token',
+                'error_description': 'No token supplied',
             });
             break;
         case ERROR_TYPES.VALIDATION_FAILURE:
             response.status(401);
-            response.set("Connection", "close");
+            response.set('Connection', 'close');
             response.json({
-                "error": "invalid_token",
-                "error_description": "The token is invalid or has expired"
+                'error': 'invalid_token',
+                'error_description': 'The token is invalid or has expired',
             });
             break;
         case ERROR_TYPES.INSUFFICIENT_SCOPE:
             response.status(403);
-            response.set("Connection", "close");
+            response.set('Connection', 'close');
             response.json({
-                "error": "insufficient_scope",
-                "error_description": "The supplied token is missing a required scope"
+                'error': 'insufficient_scope',
+                'error_description': 'The supplied token is missing a required scope',
             });
             break;
         default:
             response.status(500);
-            response.set("Connection", "close");
+            response.set('Connection', 'close');
             response.json({
-                "error": "unexpected_error",
-                "error_description": error
+                'error': 'unexpected_error',
+                'error_description': error,
             });
         }
 
         next(error, null);
         return;
-    });
+    }
+
+    try {
+        this.validateToken(config, callback);
+    } catch (exception) {
+        return callback(ERROR_TYPES.INVALID_TOKEN, null);
```

Comment:
What happens when the exception is something else? This doesn't seem right to me.